openshift/machine-config-operator
# OCPBUGS-18456,OCPBUGS-18458,OCPBUGS-18414: Configures SSH Keys and Password for core, fixes config drift degradation

Verifies that OS updates after applying image so that password and ssh keys are configurable. checks that pullspecs are different before calling rpm-ostree.



**- What I did**
The code handles updating the password hashes of the user accounts and SSH keys for users if the password configuration has changed by doing more OS updates after applying an image. It also verifies whether the oldImage pullspec is different from the newImage pullspec before performing an update with rpm-ostree.

**- How to verify it**
Password and SSH keys bugs:
1. Apply Machine configs with passwords and ssh keys
2. Verify that ssh keys are found and core can login with set password

Config Drift Bug:
1. Enable the OCB functionality for worker pool
2. Create a MC to deploy a new file
3. Modify the content of the file /etc/mco-test-file making a backup first
4. wait until the MCP reports the config drift issue
5. observe reconciliation process

**- Description for the changelog**
